lib/services/local/generic/resource153.go

```diff
@@ -0,0 +1,42 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package generic
+
+import (
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	"github.com/gravitational/teleport/api/types"
+)
+
+// resourceMetadataAdapter is an adapter for RFD 153-style resources.
+type resourceMetadataAdapter[T types.ResourceMetadata] struct {
```

Comment:
Don't we have one of those already?

Would this be better placed under API?

Reply:
`resource153ToLegacyAdapter` is not generic, which is important here, and also isn't exported.

Reply:
Not sure what you mean here?